google-home-community.groovy

```diff
@@ -4712,6 +4748,12 @@ private deleteDeviceTrait_SensorState(deviceTrait) {
     GOOGLE_SENSOR_STATES.each { sensorType, numericUnits ->
         app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.numericUnits")
     }
+    GOOGLE_SENSOR_STATES.each { sensorType, reportsDescriptiveState ->
+        app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.reportsDescriptiveState")
+    }
+    GOOGLE_SENSOR_STATES.each { sensorType, reportsNumericState ->
+        app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.reportsNumericState")
+    }    
```

Comment:
Why is this (and the other settings) looping over every sensor type multiple times?  Shouldn't this do something more like:
```
GOOGLE_SENSOR_STATES.each { sensorType, sensorSettings ->
    app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.availableStates")
    app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.descriptiveAttribute")
    app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.numericAttribute")
    app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.numericUnits")
    app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.reportsDescriptiveState")
    app.removeSetting("${deviceTrait.name}.sensorTypes.${sensorType}.reportsNumericState")
}
```

Reply:
Yes, much better.  I cleaned that up.